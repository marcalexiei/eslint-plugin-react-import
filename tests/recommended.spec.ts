import path from 'node:path';
import { fileURLToPath } from 'node:url';

import { describe, expect, it } from 'vitest';

import { loadESLint } from 'eslint';

describe('config recommended', async () => {
  const dirname = path.dirname(fileURLToPath(import.meta.url));
  const ESLint = await loadESLint({ useFlatConfig: true });
  const eslint = new ESLint({
    cwd: path.join(dirname, './fixtures/recommended'),
  });

  it('should have no error for valid file', async () => {
    const res = await eslint.lintFiles('./src/valid.jsx');

    expect(res).toHaveLength(1);

    const [validFileResult] = res;
    expect(validFileResult).toHaveProperty('errorCount', 0);
    expect(validFileResult.messages).toHaveLength(0);
  });

  it('should have errors for invalid', async () => {
    const res = await eslint.lintFiles('./src/invalid.jsx');

    expect(res).toHaveLength(1);

    const [invalidResult] = res;
    expect(invalidResult).toHaveProperty('errorCount', 2);
    expect(invalidResult).toHaveProperty('fixableErrorCount', 2);
    expect(invalidResult.messages).toMatchSnapshot(
      'should have 2 errors: 1 for import and 1 for prefix',
    );
  });

  it('should raise error when renaming a named import of `react`', async () => {
    const res = await eslint.lintFiles('./src/invalid-renamed-import.jsx');

    expect(res).toHaveLength(1);

    const [invalidResult] = res;
    expect(invalidResult).toHaveProperty('errorCount', 2);
    expect(invalidResult).toHaveProperty('fixableErrorCount', 2);
    expect(invalidResult.messages).toMatchSnapshot(
      '1 error for the import and 1 error for the useState',
    );
  });
});
