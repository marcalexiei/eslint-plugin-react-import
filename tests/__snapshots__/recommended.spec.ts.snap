// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config recommended > should have errors for invalid > should have 2 errors: 1 for import and 1 for prefix 1`] = `
[
  {
    "column": 1,
    "endColumn": 41,
    "endLine": 1,
    "fix": {
      "range": [
        0,
        40,
      ],
      "text": "import * as React from 'react';",
    },
    "line": 1,
    "message": "You should import React using namespace syntax",
    "messageId": "wrongImport",
    "nodeType": null,
    "ruleId": "react-import/syntax",
    "severity": 2,
  },
  {
    "column": 18,
    "endColumn": 26,
    "endLine": 4,
    "fix": {
      "range": [
        84,
        92,
      ],
      "text": "React.useState",
    },
    "line": 4,
    "message": "This React import should have a 'React.' prefix",
    "messageId": "addPrefix",
    "nodeType": null,
    "ruleId": "react-import/syntax",
    "severity": 2,
  },
]
`;

exports[`config recommended > should raise error when renaming a named import of \`react\` > 1 error for the import and 1 error for the useState 1`] = `
[
  {
    "column": 1,
    "endColumn": 58,
    "endLine": 1,
    "fix": {
      "range": [
        0,
        57,
      ],
      "text": "import * as React from 'react';",
    },
    "line": 1,
    "message": "You should import React using namespace syntax",
    "messageId": "wrongImport",
    "nodeType": null,
    "ruleId": "react-import/syntax",
    "severity": 2,
  },
  {
    "column": 18,
    "endColumn": 31,
    "endLine": 4,
    "fix": {
      "range": [
        101,
        114,
      ],
      "text": "React.useState",
    },
    "line": 4,
    "message": "This React import should have a 'React.' prefix",
    "messageId": "addPrefix",
    "nodeType": null,
    "ruleId": "react-import/syntax",
    "severity": 2,
  },
]
`;
